File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        int largest=INT_MIN;
        int slargest=INT_MIN;
        while(n!=0){
            int digit=n%10;
            if(digit>largest){
                slargest=largest;
                largest=digit;
            }else if(digit> slargest){
                slargest=digit;
            }
            n/=10;
        }
        return largest*slargest;
    }
};

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.java
class Solution {
    public int maxProduct(int n) {
        int largest=Integer.MIN_VALUE;
        int slargest=Integer.MIN_VALUE;

        while(n!=0){
            int lastdigit=n%10;
            if(lastdigit>largest){
                slargest=largest;
                largest=lastdigit;
            }else if(lastdigit>slargest){
                slargest=lastdigit;
            }
            n/=10;
        }
        return largest*slargest;
    }
}